sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java

```diff
@@ -196,14 +197,15 @@ Mono<StoreResponse> writePrivateAsync(
                 try {
                     primaryURI.set(primaryUri);
                     if ((this.useMultipleWriteLocations || request.getOperationType() == OperationType.Batch) &&
-                        RequestHelper.getConsistencyLevelToUse(this.serviceConfigReader, request) == ConsistencyLevel.SESSION) {
+                        (RequestHelper.getReadConsistencyStrategyToUse(this.serviceConfigReader, request)
```

Comment:
I had a question here.

My understanding is for `Batch` we'd like to use `ReadConsistencyStrategy` since it can encapsulate reads but should we instead use `ConsistencyLevel` for non-`Batch`?

Reply:
No need - Batch is always executed as a write Operation (so follows the write routing logic independent of whether the batch only contains reads or not). So, Batch has its own consistency rules - similar to stored procedures before.